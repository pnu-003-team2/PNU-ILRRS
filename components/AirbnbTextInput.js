import React, { Component } from 'react';
import { View, Text, TextInput, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

export default class AirbnbTextInput extends Component {

  state = {
    isFocus: false,
  };

  static propTypes = {
    style: PropTypes.object,
    placeholder: PropTypes.string.isRequired,
    onChangeText: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired,
  };
  
  static defaultProps = {
    style: null,
  };

  onTextChange = (text) => {
    this.props.onChangeText(text);
  }

  onFocus = () => {
    this.setState({ isFocus: true });
  }

  onBlur = () => {
    this.setState({ isFocus: false });
  }

  render() {
    const { value } = this.props;
    const { isFocus } = this.state;
    return (
      <View style={styles.container}>
        { (!isFocus && value.length === 0) ? <Text style={styles.placeholder}>{this.props.placeholder}</Text> : null}
        <TextInput
          style={styles.textInput}
          value={value}
          onChangeText={this.onTextChange}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
  },
  placeholder: {
    position: 'absolute',
    top: 2,
  },
  textInput: {
    fontSize: 24,
    paddingVertical: 10,
    borderBottomWidth: 1,
    borderBottomColor: 'black',
  },
});
