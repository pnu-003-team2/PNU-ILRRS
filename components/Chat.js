import React from 'react';
import { GiftedChat } from 'react-native-gifted-chat';

export default class Chat extends React.Component {
  state = {
    messages: [],
  };

  componentWillMount() {
    this.setState({
      messages: [
        {
          _id: 1,
          text: 'Hello developer',
          createdAt: new Date(),
          user: {
            _id: 2,
            name: 'React Native',
            avatar: 'https://placeimg.com/140/140/any',
          },
        },
      ],
    });
  }

  handleSend = (messages = []) => {
    this.setState(previousState => ({
      messages: GiftedChat.append(previousState.messages, messages),
    }));
  }

  render() {
    return (
      <GiftedChat
        locale="ko"
        messages={this.state.messages}
        onSend={this.handleSend}
        user={{
          _id: 1,
        }}
      />
    );
  }
}
