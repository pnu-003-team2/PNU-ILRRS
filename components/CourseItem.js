import React from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
  View,
} from 'react-native';

import SeperatedText from '../components/SeperatedText';
import RoundLabel from '../components/RoundLabel';

const propTypes = {
  course: PropTypes.shape({
    className: PropTypes.string,
    classDivision: PropTypes.string,
    code: PropTypes.number,
    id: PropTypes.number.isRequired,
    professorName: PropTypes.string,
    timeTable: PropTypes.string,
  }).isRequired,
  onPress: PropTypes.func,
};

const defaultProps = {
  onPress() {},
};

const CourseItem = ({ course, onPress }) => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.course}>
      <View style={styles.courseTopView}>
        <View style={styles.professorBox}>
          <Text >{course.professorName} 교수님</Text>
        </View>
        <RoundLabel color={'#4f54fb'} text={course.classDivision} />
        <RoundLabel color={'#de7924'} text={course.code + ' 분반'} />
      </View>

      <View style = {styles.courseBottomView}>
        <Text style={[styles.courseTitleText, styles.bottomPadding]}>{course.className}</Text>
        <SeperatedText style={styles.coureTimeText} timeTable={course.timeTable}/>
      </View>
    </View>
  </TouchableWithoutFeedback>
);

const styles = StyleSheet.create({
  course: {
    flexDirection: 'column',
    alignItems: 'stretch',
    borderRadius: 15,
    backgroundColor : 'white',
    paddingLeft: 20,
    paddingRight: 15,
    paddingVertical: 15,
    marginTop: 20,
    marginHorizontal : 30,
    //shadowOffset: '10',
    //elevation: 50,

  },

  courseTopView: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 8,
    //backgroundColor: 'red',
  },
  professorBox: {
    flex: 3,
    justifyContent: 'flex-end',
  },
  professorText: {
    fontSize: 14,
    color: '#404040',
    fontFamily: 'Roboto-Regular',
  },

  courseBottomView:{
    flex:1,
    //backgroundColor: 'blue',
  },
  courseTitleText: {
    fontFamily: 'Roboto-Bold',
    fontSize: 20,
  },
  coureTimeText:{
    fontFamily: 'Roboto-Regular',
    fontSize: 14,
  },
  bottomPadding:{
    paddingBottom: 7,
  },
});

CourseItem.propTypes = propTypes;
CourseItem.defaultProps = defaultProps;

export default CourseItem;
