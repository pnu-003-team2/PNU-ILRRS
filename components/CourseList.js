import React from 'react';
import PropTypes from 'prop-types';
import { FlatList, StyleSheet, Text, View } from 'react-native';


function SepearatedText({ timeTable }) {
  let [firstLine,secondLine] = timeTable.split(',',2);
  
  return (
    <View>
      <Text>{firstLine}</Text>
      <Text>{secondLine}</Text>
    </View>
  );
}


const propTypes = {
  courses: PropTypes.arrayOf(PropTypes.shape({
    className: PropTypes.string,
    classDivision: PropTypes.string,
    code: PropTypes.number,
    professorName: PropTypes.string,
    timeTable: PropTypes.string,
  })),
};

const defaultProps = {
  courses: [],
};

function CourseList({ courses }) {
  return (
    <FlatList
      data={courses.map(course => ({ ...course, key: course.id.toString() }))}
      renderItem={({ item }) => (
        <View style={styles.course}>
          <View style={styles.coureLeftView}>
            <Text style={[styles.titleText, styles.bottomPadding]}>{item.className}</Text>
            <Text style={item.classDivision.includes('전공') ? [styles.redText,styles.bottomPadding] : [styles.grayText,styles.bottomPadding]}>{item.classDivision}</Text>
            <Text style={[styles.codeText]}>{item.code} 분반</Text>
            <SepearatedText timeTable={item.timeTable}/>
          </View >
          <View style={styles.courseRightView}>
            <Text style={styles.professorNameText}>{item.professorName} 교수님</Text>
          </View>
        </View>
      )}
    />
  );
}


const styles = StyleSheet.create({
  course: {
    flexDirection: 'row',
    alignItems: 'stretch',
    padding: 16,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: 'skyblue',
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 10,
  },
  coureLeftView: {
    flex: 3,
    flexDirection: 'column',
  },
  courseRightView:{
    flex: 1,

  },
  professorNameText: {
    fontSize: 13,
  },
  titleText: {
    fontWeight: '600',
    fontSize: 19,
  },
  bottomPadding: {
    paddingBottom : 16,
  },
  redText: {
    color:  'red',
  },
  grayText: {
    color:  'gray',
  },
  codeText: {
    color : 'gray',
    paddingBottom: 4,
  },
});

CourseList.propTypes = propTypes;
CourseList.defaultProps = defaultProps;
export default CourseList;

