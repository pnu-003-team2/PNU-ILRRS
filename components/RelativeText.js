import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text } from 'react-native';

const propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.object,
  widthPercent: PropTypes.number,
};

const defaultProps = {
  style: null,
  widthPercent: 80,
};

function RelativeText({
  children,
  style,
  widthPercent,
  ...rest
}) {
  if (!children) {
    return null;
  }

  return (
    <Text
      autoCapitalize="none"
      children={children}
      style={[styles.text, { width: `${widthPercent}%` }, style]}
      {...rest}
    />
  );
}

const styles = StyleSheet.create({
  text: {
    marginBottom: 15,
  },
});

RelativeText.propTypes = propTypes;
RelativeText.defaultProps = defaultProps;

export default RelativeText;
