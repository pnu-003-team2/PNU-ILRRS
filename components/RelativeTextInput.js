import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, TextInput } from 'react-native';

const propTypes = {
  style: PropTypes.object,
  widthPercent: PropTypes.number,
};

const defaultProps = {
  style: null,
  widthPercent: 80,
};

function RelativeTextInput({
  style,
  widthPercent,
  ...rest
}) {
  return (
    <TextInput
      autoCapitalize="none"
      style={[styles.textInput, { width: `${widthPercent}%` }, style]}
      {...rest}
    />
  )
}

const styles = StyleSheet.create({
  textInput: {
    borderWidth: 1,
    borderColor: 'gray',
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    borderBottomLeftRadius: 5,
    borderBottomRightRadius: 5,
    marginBottom: 10,
    padding: 10,
    height: 40,
  },
})

RelativeTextInput.propTypes = propTypes;
RelativeTextInput.defaultProps = defaultProps;

export default RelativeTextInput;
