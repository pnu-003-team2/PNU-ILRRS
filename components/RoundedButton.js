import React, { Component } from 'react';
import { TouchableOpacity ,View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

export default class RoundedButton extends Component {

  static propTypes = {
    onPress: PropTypes.func.isRequired,
    buttonText: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    buttonColor: PropTypes.string.isRequired,
  };

  render() {
      const {buttonText, onPress, disabled, buttonColor} = this.props;
    return (
        <TouchableOpacity onPress={onPress} disabled={disabled}>
            <View style={[styles.container,{backgroundColor: buttonColor}]}>
                <Text style={styles.textView}>{buttonText}</Text>
            </View>
        </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 65,
    width: 165,
    borderRadius: 32.5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textView: {
    color: '#fafafa',
    fontSize: 25,
    fontFamily: 'Roboto-Bold',
  },

});
