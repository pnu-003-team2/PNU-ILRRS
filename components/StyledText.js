import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

export const HelveticaText = props => (
  <Text
    {...props}
    style={[props.style, { fontFamily: 'Helvetica' }]}
  />
);
HelveticaText.propTypes = { style: PropTypes.object };
HelveticaText.defaultProps = { style: undefined };
