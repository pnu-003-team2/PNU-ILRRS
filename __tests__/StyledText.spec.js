import 'react-native';
import React from 'react';
import { render } from 'react-native-testing-library';

import { HelveticaText } from '../components/StyledText';

describe('<StyledText />', () => {
  it('should renders <HelveticaText /> correctly', () => {
    const { toJSON } = render(<HelveticaText>Test</HelveticaText>);
    expect(toJSON()).toMatchSnapshot();
  });
});
