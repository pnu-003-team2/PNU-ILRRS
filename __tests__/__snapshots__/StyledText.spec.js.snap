// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledText /> should renders <HelveticaText /> correctly 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "Helvetica",
      },
    ]
  }
>
  Test
</Text>
`;

exports[`<StyledText /> should renders <MonoText /> correctly 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Test
</Text>
`;
